### PR TITLE
fix: add serial format limit to fix OOM issue when formatting a few disks in parallel in csi-azuredisk-node

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/king bug

**What this PR does / why we need it**:

bug: add serial format limit to fix OOM issue when formatting a few disks in parallel in csi-azuredisk-node
add flags `max-concurrent-format` and `concurrent-format-timeout` as default 2 and 2 minutes

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
fix: add serial format limit to fix OOM issue when formatting a few disks in parallel in csi-azuredisk-node
```

### CODE REVIEW

File: pkg/azuredisk/azuredisk_option.go [hunk 1]
```diff
@@ -103,6 +105,8 @@ func (o *DriverOptions) AddFlags() *flag.FlagSet {
 	fs.BoolVar(&o.DisableAVSetNodes, "disable-avset-nodes", false, "disable DisableAvailabilitySetNodes in cloud config for controller")
 	fs.BoolVar(&o.RemoveNotReadyTaint, "remove-not-ready-taint", true, "remove NotReady taint from node when node is ready")
 	fs.StringVar(&o.Endpoint, "endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	fs.Int64Var(&o.MaxConcurrentFormat, "max-concurrent-format", 3, "maximum number of concurrent format exec calls")
+	fs.Int64Var(&o.ConcurrentFormatTimeout, "concurrent-format-timeout", 60, "maximum time in seconds duration of a format operation before its concurrency token is released")
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk_option.go:109]: set as 2min

**Review by andyzhangx — COMMENTED** [review]
pls also squash all commits, thx

File: pkg/azuredisk/azuredisk_option.go [hunk 2]
```diff
@@ -103,6 +105,8 @@ func (o *DriverOptions) AddFlags() *flag.FlagSet {
 	fs.BoolVar(&o.DisableAVSetNodes, "disable-avset-nodes", false, "disable DisableAvailabilitySetNodes in cloud config for controller")
 	fs.BoolVar(&o.RemoveNotReadyTaint, "remove-not-ready-taint", true, "remove NotReady taint from node when node is ready")
 	fs.StringVar(&o.Endpoint, "endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	fs.Int64Var(&o.MaxConcurrentFormat, "max-concurrent-format", 3, "maximum number of concurrent format exec calls")
```
[the remark this answers is not in the document]
Comment (umagnus) [reply pkg/azuredisk/azuredisk_option.go]: done

File: pkg/azuredisk/azuredisk_option.go (hunk 1 above)
Comment (umagnus) [reply pkg/azuredisk/azuredisk_option.go]: done

File: pkg/mounter/safe_mounter_windows.go [hunk 3]
```diff
@@ -411,7 +412,7 @@ func newCSIProxyMounter() (*csiProxyMounter, error) {
 	}, nil
 }
 
-func NewSafeMounter(enableWindowsHostProcess, useCSIProxyGAInterface bool) (*mount.SafeFormatAndMount, error) {
+func NewSafeMounter(enableWindowsHostProcess, useCSIProxyGAInterface bool, maxConcurrentFormat int, concurrentFormatTimeout time.Duration) (*mount.SafeFormatAndMount, error) {
 	if enableWindowsHostProcess {
 		klog.V(2).Infof("using windows host process mounter")
 		return &mount.SafeFormatAndMount{
```
Comment (andyzhangx) [inline pkg/mounter/safe_mounter_windows.go:418]: pls also make the same changes on windows host process

Comment (umagnus) [reply pkg/mounter/safe_mounter_windows.go:418]: Should we also use it in csiproxyGA nad csiproxyv1beta?

Comment (andyzhangx) [reply pkg/mounter/safe_mounter_windows.go]: not necessary since we are using windows host process deployment only now.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**